frontend/src/components/widgets/inputs/SwatchPairInput.vue

```diff
@@ -135,20 +136,16 @@ export default defineComponent({
 		};
 	},
 	mounted() {
-		registerResponseHandler(ResponseType.UpdateWorkingColors, (responseData: Response) => {
-			const colorData = responseData as UpdateWorkingColors;
-			if (!colorData) return;
+		registerJsMessageHandler(UpdateWorkingColors, (colorData) => {
 			const { primary, secondary } = colorData;
 
-			this.primaryColor = { r: primary.red, g: primary.green, b: primary.blue, a: primary.alpha };
-			let color = this.primaryColor;
+			this.primaryColor = primary.toRgb();
 			let button = this.getRef<HTMLButtonElement>("primaryButton");
-			button.style.setProperty("--swatch-color", `rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`);
+			button.style.setProperty("--swatch-color", primary.toString());
 
-			this.secondaryColor = { r: secondary.red, g: secondary.green, b: secondary.blue, a: secondary.alpha };
-			color = this.secondaryColor;
+			this.secondaryColor = secondary.toRgb();
```

Comment:
Might as well group this line together with the `this.primaryColor = primary.toRgb();` line above.